pandas/tests/indexes/common.py

```diff
@@ -635,6 +635,16 @@ def test_delete_base(self):
                 # either depending on numpy version
                 result = idx.delete(len(idx))
 
+    def test_equals(self):
+
+        for name, idx in compat.iteritems(self.indices):
+            self.assertTrue(idx.equals(idx))
+            self.assertTrue(idx.equals(idx.copy()))
+            self.assertTrue(idx.equals(idx.astype(object)))
+            self.assertFalse(idx.equals(list(idx)))
```

Comment:
agreed ok with coercing an Index but not anything else (ndarray, list, Series)

so much can go wrong with auto coercion in this kind of test